backend/routes/shipping-zones.js

```diff
@@ -1,84 +1,38 @@
 const fs = require('fs')
 const path = require('path')
-const pick = require('lodash/pick')
-const snakeCase = require('lodash/snakeCase')
 
 const { authSellerAndShop, authRole } = require('./_auth')
 const { DSHOP_CACHE } = require('../utils/const')
+const { setConfig, getConfig } = require('../utils/encryptedConfig')
 
 module.exports = function (router) {
   router.put(
     '/shipping-zones',
     authSellerAndShop,
     authRole('admin'),
     async (req, res) => {
-      const { shippingZones } = req.body
-
-      try {
-        const newData = shippingZones.map((zone) => ({
-          ...zone,
-          id: zone.id || snakeCase(zone.label)
-        }))
-
-        const outDir = path.resolve(`${DSHOP_CACHE}/${req.shop.authToken}/data`)
-        const shippingPath = `${outDir}/shipping.json`
-        fs.writeFileSync(shippingPath, JSON.stringify(newData, undefined, 2))
-
-        await req.shop.update({
-          hasChanges: true
-        })
-
-        res.send({ success: true })
-      } catch (e) {
-        res.json({ success: false })
-      }
-    }
-  )
-
-  router.put(
-    '/shipping-zones/:shippingId',
-    authSellerAndShop,
-    authRole('admin'),
-    async (req, res) => {
-      const { shippingId } = req.params
+      const { destinations, shippingFrom, processingTime } = req.body
 
       try {
         const outDir = path.resolve(`${DSHOP_CACHE}/${req.shop.authToken}/data`)
         const shippingPath = `${outDir}/shipping.json`
-        const existingData = JSON.parse(fs.readFileSync(shippingPath))
-
-        const newZoneData = {
-          ...pick(req.body, [
-            'label',
-            'detail',
-            'countries',
-            'amount',
-            'processingTime'
-          ]),
-          id: shippingId
-        }
-
-        let existingZone = false
-        const shippingZones = existingData.map((zone) => {
-          if (zone.id === shippingId) {
-            existingZone = true
-            return newZoneData
-          }
-
-          return zone
-        })
-
-        if (!existingZone) {
-          shippingZones.push(newZoneData)
-        }
-
         fs.writeFileSync(
           shippingPath,
-          JSON.stringify(shippingZones, undefined, 2)
+          JSON.stringify(destinations, undefined, 2)
+        )
+
+        const newConfig = setConfig(
+          {
+            ...getConfig(req.shop.config),
```

Comment:
Should we pick the specific shipping related fields coming from the request that need to be persisted?
This is to avoid a situation where somehow the UI sends an invalid config in the request (due to a bug) and the whole shop configuration gets overwritten.

Reply:
`req.shop.config` is actually from DB. Only the `shippingFrom` and `processingTime` are picked from body